### ISSUE TITLE
Karazhan: boss terestian illhoof

### ISSUE THREAD

**Issue by amaroc** [OregonCore/OregonCore#1539]
**Revision**:
OregonCore Rev: 3602

**Operating system**:
ubnt 16.04

**Current behavior**: 
When players get in "chains", another players in raid, cant target that chains and attack them.
So player just died (heals cant heal him all the time).

**Expected behavior**: 
Damagedealers, must switch target on "Chains" and can attack it and kill it, before player died.

**Steps to reproduce the problem**:
.tele karazhan
http://wowhead.com/npc=15688 - go this boss in 2 windows, and check this.

**Comment by Selphius** [comment]
Tested it on Clean OregonCore, can't reproduce it. I can normally attack the chains from other window.

**Comment by ljwlaji** [comment]
try to set the Chains faction to "14" in sciript cpp after the Chains has been created

**Comment by Selphius** [comment]
@ljwlaji Isn't faction 14 made as default for them on Clean core? Well obviously he can try/check that, maybe he tinkered with faction on this creature at some point.

On side note I wrote fix for this entire boss yesterday, it's awaiting pull check, so you can try it, maybe it will help.

**Comment by amaroc** [comment]
Very strange, now it appears, maybe before i didn`t see  "Demon chains" because it appears near boss.

It must be in center of room, in center of green rune. With "teleported" player. So with faction all good. I can target it and kill it.

And another question why there 2 Kil'Rek (big imps) .

**Comment by Selphius** [comment]
You checked it on my commit for this boss and it works fine? or still on old one.

Ah right :P. I fixed the Kil'rek so he is going to be summoned during battle. In battle start after 5 seconds, and later when killed 30s after he died. What I forgot is that I manually deleted the one already standing near Terestian when he shouldn't be there in the first place, cause of that I forgot to include SQL for removing him.

Edit: Gonna commit it later with more SQL fixes.

**Comment by robinsch** [comment]
https://github.com/diatlov/tbcpvp-test/commit/6ffbecd04caf8363313b9323073d52f8f49978ad

**Comment by ljwlaji** [comment]
@Selphius  yep 14 Is Hostile to All Player Race Faction in Clean Core
But To Stable Change,You Can Edit To Boss's faction aslo Like setFaction(boss->GetFaction());

**Comment by amaroc** [comment]
@Selphius i cheked it on old OregonCore Rev: 3602, it just appears not in center of room.
Maybe after your fix, chains appears in center.

**Comment by Selphius** [comment]
Well yeah, since I made it appear on top of player. Still some work needs to be done to those scripts -- mostly fixing spell for portals. 
Also should there be a limit to imps? or should they spawn indifinitely.

**Comment by DanielBallaSZTE** [comment]
@Selphius They should spawn indefinitely, but I don't know you have like 5000ms declared in the core if I recall right, should change that value to 10000, according to the wowwiki. Also the spawning begins 10s after the fight begun. Kil'rek also should respawn after 45s, don't know if that's implemented.

**Comment by Selphius** [comment]
@DanielBallaSZTE 
Well I used, pre-nerf values prior to 2.0. Timers have been increased in later patches nerfing the boss at the same time. Kil'rek has implemented 5s into the fight he is summoned, 30s after his death to respawn.
Imps begun spawning after 10s+ into the fight, when portal have been summoned, new imps come out after (1.5s - 5s)

Seems like 1 portal should summon MAX 10 imps. so 20 imps in total should be in the fight at the same time. Checked Corecraft video.
https://www.youtube.com/watch?v=AFT_tPUp5yA
